subsys/storage/stream/stream_flash.c

```diff
@@ -189,6 +212,11 @@ int stream_flash_init(struct stream_flash_ctx *ctx, struct device *fdev,
 		return -EFAULT;
 	}
 
+	if (buf_len % flash_get_write_block_size(fdev)) {
```

Comment:
Can we do that check at the beginning of `stream_flash_init`, it looks like it is parameter check.
The `buf_len` is also used at line 200, so if it is incorrect here, then it is also incorrect at that line.